tree/ntuple/v7/src/RField.cxx

```diff
@@ -895,6 +919,27 @@ ROOT::Experimental::RClassField::RClassField(std::string_view fieldName, std::st
       Attach(std::move(subField),
 	     RSubFieldInfo{kDataMember, static_cast<std::size_t>(dataMember->GetOffset())});
    }
+
+   // Add post-read callbacks for I/O customization rules; only rules that target transient members are allowed for now
+   // TODO(jalopezg): revise after supporting schema evolution
+   if (const auto ruleset = fClass->GetSchemaRules()) {
+      auto referencesNonTransientMembers = [this](const ROOT::TSchemaRule *rule) {
```

Comment:
I think I'd find it more readable to make this a private method of `RClassField`

Reply:
Hmmm... as part of the changes to filter rules according to class version, this code has been moved to `RegisterReadCallbacks()`.  In its current state, I find the code fairly readable, but let me know if you still prefer it as a private member function.